openedx/core/djangoapps/content_libraries/collections/rest_api/v1/views.py

```
@@ -0,0 +1,181 @@
+"""
+Collections API Views
+"""
+
+from __future__ import annotations
+
+from django.http import Http404
+
+from rest_framework.response import Response
+from rest_framework.viewsets import ModelViewSet
+from rest_framework.status import HTTP_405_METHOD_NOT_ALLOWED
+
+from opaque_keys.edx.locator import LibraryLocatorV2
+
+from openedx_events.content_authoring.data import LibraryCollectionData
+from openedx_events.content_authoring.signals import (
+    LIBRARY_COLLECTION_CREATED,
+    LIBRARY_COLLECTION_UPDATED,
+)
+
+from openedx.core.djangoapps.content_libraries import api, permissions
+from openedx.core.djangoapps.content_libraries.serializers import (
+    ContentLibraryCollectionSerializer,
+    ContentLibraryCollectionCreateOrUpdateSerializer,
+)
+
+from openedx_learning.api.authoring_models import Collection
+from openedx_learning.api import authoring as authoring_api
+
+
+class LibraryCollectionsView(ModelViewSet):
+    """
+    Views to get, create and update Library Collections.
+    """
+
+    serializer_class = ContentLibraryCollectionSerializer
+
+    def _verify_and_fetch_library_collection(self, library_key, collection_id, user, permission) -> Collection | None:
+        """
+        Verify that the collection belongs to the library and the user has the correct permissions
+        """
+        try:
+            library_obj = api.require_permission_for_library_key(library_key, user, permission)
+        except api.ContentLibraryNotFound as exc:
+            raise Http404 from exc
+
+        collection = None
```

If we use the `convert_exceptions` decorator and change the API, we can just make it throw if not found.